icu4c/source/common/mlbe.cpp

```diff
@@ -128,135 +122,169 @@ int32_t MlBreakEngine::divideUpRange(UText *inText, int32_t rangeStart, int32_t
-        score += (2 * fModel.geti(feature.setTo(u"UW2:", 4).append(elementList[1])));
+    if (indexList[1] != -1) {
+        // UW2
+        end = (indexList[2] != -1) ? indexList[2] : codeUts;
+        score += fModel[1].geti(inString.tempSubString(indexList[1], end - indexList[1]));
     }
-    if (elementList[2] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW3:", 4).append(elementList[2])));
+    if (indexList[2] != -1) {
+        // UW3
+        end = (indexList[3] != -1) ? indexList[3] : codeUts;
+        score += fModel[2].geti(inString.tempSubString(indexList[2], end - indexList[2]));
     }
-    if (elementList[3] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW4:", 4).append(elementList[3])));
+    if (indexList[3] != -1) {
+        // UW4
+        end = (indexList[4] != -1) ? indexList[4] : codeUts;
+        score += fModel[3].geti(inString.tempSubString(indexList[3], end - indexList[3]));
     }
-    if (elementList[4] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW5:", 4).append(elementList[4])));
+    if (indexList[4] != -1) {
+        // UW5
+        end = (indexList[5] != -1) ? indexList[5] : codeUts;
+        score += fModel[4].geti(inString.tempSubString(indexList[4], end - indexList[4]));
     }
-    if (elementList[5] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW6:", 4).append(elementList[5])));
+    if (indexList[5] != -1) {
+        // UW6
+        end = codeUts;
+        score += fModel[5].geti(inString.tempSubString(indexList[5], end - indexList[5]));
     }
-    if (elementList[1] != INVALID && elementList[2] != INVALID) {
-        score += (2 * fModel.geti(
-                          feature.setTo(u"BW1:", 4).append(elementList[1]).append(elementList[2])));
+    if (indexList[1] != -1 && indexList[2] != -1) {
+        // BW1
+        end = (indexList[3] != -1) ? indexList[3] : codeUts;
+        score += fModel[6].geti(inString.tempSubString(indexList[1], end - indexList[1]));
     }
-    if (elementList[2] != INVALID && elementList[3] != INVALID) {
-        score += (2 * fModel.geti(
-                          feature.setTo(u"BW2:", 4).append(elementList[2]).append(elementList[3])));
+    if (indexList[2] != -1 && indexList[3] != -1) {
+        // BW2
+        end = (indexList[4] != -1) ? indexList[4] : codeUts;
+        score += fModel[7].geti(inString.tempSubString(indexList[2], end - indexList[2]));
     }
-    if (elementList[3] != INVALID && elementList[4] != INVALID) {
-        score += (2 * fModel.geti(
-                          feature.setTo(u"BW3:", 4).append(elementList[3]).append(elementList[4])));
+    if (indexList[3] != -1 && indexList[4] != -1) {
+        // BW3
+        end = (indexList[5] != -1) ? indexList[5] : codeUts;
+        score += fModel[8].geti(inString.tempSubString(indexList[3], end - indexList[3]));
     }
-    if (elementList[0] != INVALID && elementList[1] != INVALID && elementList[2] != INVALID) {
```

Comment:
after your last PR, BW1 to BW3  could be merged into one for loop
```
for (int32_t j = 0; j < 3; j++) {
  if (indexList[j+1] != -1 && indexList[j+2] != -1) {
        // BW(j+1)
        end = (indexList[j+3] != -1) ? indexList[j+3] : codeUts;
        score += fModel[j+6].geti(inString.tempSubString(indexList[j+1], end - indexList[j+1]));
    }
}
```

Reply:
Done